### PR TITLE
Import in Signup: Add preview step to import-onboarding flow

### PR DESCRIPTION
**Pull request by andfinally**
#### Changes proposed in this Pull Request

* This PR adds a "preview" step to the import signup flow. It's intended to set expectations about what the user can expect at the end of the import process. We plan to show it after someone has entered the URL of a site they want to import. We should have detected the site type by that stage.
* Based on the type of site detected, we show a basic list of features we'll import, and a sentence about what we can't import.

image

* At this stage, we'll show the icon for the service (for example GoDaddy) with the site title and URL instead of the site's favicon. This is because not all favicons are big enough to show here. We'll need to deal with that in a separate PR.

#### Testing instructions

* Apply the PR and go to http://calypso.localhost:3000/start/import-onboarding/, or use `calypso.live`.
* Click the `Already have a website?` link on the `site-type` step to go to the `from-url` step.
* Enter the URL of a GoDaddy or Wix site to import from, for example https://omowale.org or https://www.totalsportswomensrace.com.
* You should see the preview. Clicking the "Continue" button should take you to the `domains` step and ultimately to the GoDaddy or Wix importer, where the site preview should load. (We'll look at skipping that preview in another PR.)
* Check this new step only appears in the `import-onboarding` flow and not in the existing import flow from `/start/import`.

* Start again at `/start/import-onboarding/` and select one of the site types, like "Blog".
* You should to the next step in the main onboarding flow, `/start/site-topic-with-preview`.

Fixes https://github.com/Automattic/samus-private/issues/198

### CODE REVIEW

**Comment by gwwar** [comment]
Is there more context for this change?

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php [hunk 1]
```diff
@@ -191,12 +191,12 @@ public function register_blocks() {
...
-				'attributes'      => [
-					'className' => [
+				'attributes'      => array(
+					'className' => array(
 						'default' => '',
 						'type'    => 'string',
-					],
-				],
+					),
+				),
```
Comment (gwwar) [inline apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php]: Why all the janitorial changes here?

Comment (gwwar) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php]: There doesn't appear to be any other changes to the Full Site Editing plugin? Do you mind dropping the related commits?

Comment (vindl) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php]: +1 For dropping these changes, we've been using array shorthand syntax throughout the FSE plugin on purpose. For context, our minimum plugin requirements are PHP 5.6+.

Comment (andfinally) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php]: Thanks @gwwar and @vindl, that change should not have been included in this draft – it's now removed.

File: client/signup/steps/import-preview/index.jsx [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+	constructor( props ) {
+		super( props );
+
+		const { translate } = this.props;
+		this.previewData = {
+			wordpress: {
+				supported: [
+					translate( 'Posts' ),
+					translate( 'Pages' ),
+					translate( 'Images and other media' ),
```
Comment (a8ci18n) [inline client/signup/steps/import-preview/index.jsx]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Finding Free Images and other Media' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/importer/logos/godaddy-gocentral.jsx [hunk 3]
```diff
@@ -5,11 +5,14 @@
...
 import React from 'react';
 
-export default function GoDaddyGoCentralLogo( { size } ) {
+export default function GoDaddyGoCentralLogo( {
+	size,
+	className = 'godaddy-gocentral social-logo importer__service-icon',
+} ) {
 	return (
 		<img
 			// eslint-disable-next-line wpcalypso/jsx-classname-namespace
```
Comment (mattsherman) [inline client/my-sites/importer/logos/godaddy-gocentral.jsx]: Do we still need the `eslint-disable-next-line` here?

File: client/my-sites/importer/logos/medium.jsx [hunk 4]
```diff
@@ -5,10 +5,10 @@
  */
 import React from 'react';
 
-export default () => (
+export default ( { className = 'medium-logo social-logo importer__service-icon' } ) => (
 	<svg
 		/*eslint-disable wpcalypso/jsx-classname-namespace */
```
Comment (mattsherman) [inline client/my-sites/importer/logos/medium.jsx]: Do we still need these `eslint-disable`s? If so, should replace with `eslint-disable-next-line` to tidy things up a bit.

File: client/my-sites/importer/importer-logo.jsx [hunk 5]
```diff
@@ -19,34 +19,37 @@ import SocialLogo from 'components/social-logo';
  */
 import './importer-logo.scss';
 
-const ImporterLogo = ( { icon } ) => {
+const ImporterLogo = ( { icon, className = 'importer__service-icon' } ) => {
+	/* eslint-disable wpcalypso/jsx-classname-namespace */
+
+	if ( 'blogger' === icon ) {
```
Comment (mattsherman) [inline client/my-sites/importer/importer-logo.jsx]: Why this special handling for `blogger`?

File: client/my-sites/importer/importer-logo.jsx [hunk 6]
```diff
@@ -19,34 +19,37 @@ import SocialLogo from 'components/social-logo';
...
+	}
+
 	if ( includes( [ 'wordpress', 'blogger-alt', 'squarespace' ], icon ) ) {
-		return <SocialLogo className="importer__service-icon" icon={ icon } size={ 48 } />;
+		return <SocialLogo className={ className } icon={ icon } size={ 48 } />;
 	}
 
 	if ( 'wix' === icon ) {
-		return <WixLogo />;
+		return <WixLogo className={ className } />;
```
Comment (mattsherman) [inline client/my-sites/importer/importer-logo.jsx]: Why this change? (and similar ones for other services)

File: client/my-sites/importer/importer-logo.jsx [hunk 7]
```diff
@@ -19,34 +19,37 @@ import SocialLogo from 'components/social-logo';
...
 	}
 
 	if ( 'godaddy-gocentral' === icon ) {
-		return <GoDaddyGoCentralLogo size={ 48 } />;
+		return <GoDaddyGoCentralLogo size={ 48 } className={ className } />;
```
Comment (mattsherman) [inline client/my-sites/importer/importer-logo.jsx]: Why is GoDaddy the only one that has a size specified?

File: client/my-sites/importer/logos/godaddy-gocentral.jsx [hunk 8]
```diff
@@ -5,11 +5,14 @@
...
-			className="godaddy-gocentral social-logo importer__service-icon"
+			className={ className }
```
Comment (mattsherman) [inline client/my-sites/importer/logos/godaddy-gocentral.jsx]: Why this change?

**Comment by mattsherman** [comment]
@andfinally In addition to addressing the comments, please make sure to rebase this against the latest in master to make sure we pick up the latest changes made in #35356.

I'll give this another spin once it is ready!

File: client/my-sites/importer/importer-logo.jsx (hunk 5 above)
Comment (jonathansadowski) [reply client/my-sites/importer/importer-logo.jsx]: Instead of `icon={ importSiteEngine }`, we can get this mapping (that the icon for `blogger` is `blogger-alt`) from `client/lib/importer/importer-config`. It doesn't seem right to map engine to the actual icon configured for the engine within this component.

File: client/my-sites/importer/logos/godaddy-gocentral.jsx (hunk 3 above)
Comment (andfinally) [reply client/my-sites/importer/logos/godaddy-gocentral.jsx]: The linter is happy with it removed, so I've taken it out.

File: client/my-sites/importer/logos/medium.jsx (hunk 4 above)
Comment (andfinally) [reply client/my-sites/importer/logos/medium.jsx]: The linter complains when I try to remove this one, so I've changed it to a single line comment.

File: client/my-sites/importer/importer-logo.jsx (hunk 5 above)
Comment (andfinally) [reply client/my-sites/importer/importer-logo.jsx]: Meeps, this was because we have two icons for Blogger, and use the version called `blogger-alt`. Thanks Jon, now changed.

File: client/my-sites/importer/importer-logo.jsx (hunk 6 above)
Comment (andfinally) [reply client/my-sites/importer/importer-logo.jsx]: I was hoping to keep the markup tidy by giving these components classes that were consistent with their parents in the import preview. I also thought it would help avoid having to override existing CSS. But the change made the components less simple, and it seems the linter has no problem with me using the existing classnames, so I've removed these params, thanks!

File: client/my-sites/importer/importer-logo.jsx (hunk 7 above)
Comment (andfinally) [reply client/my-sites/importer/importer-logo.jsx]: I'm not sure why a size param was implemented in this particular icon but not in the others – I've left it in and given it a default rather than lose the extra flexibility it brings.

File: client/my-sites/importer/logos/godaddy-gocentral.jsx (hunk 8 above)
Comment (andfinally) [reply client/my-sites/importer/logos/godaddy-gocentral.jsx]: Please see response in `importer-logo.jsx`.

**Comment by andfinally** [comment]
I'm not sure why we were skipping the domains step, but we were already doing this in the `import` and `import-onboarding` flows – until recently, when https://github.com/Automattic/wp-calypso/pull/35175 was merged, and now we're getting the step again.

Thanks for spotting the issue with the logos on the importer page! With the latest treatment of the icon CSS classes this is now fixed.

I've rebased the branch.

**Comment by mattsherman** [comment]
Thanks for making the updates @andfinally. Once #35409 is merged, please rebase against master so that we can see how the flow looks with that in place. I'll give this a final review/test then!

**Comment by andfinally** [comment]
Thanks @mattsherman, the branch is now rebased.

File: client/signup/steps/import-preview/style.scss [hunk 9]
```diff
@@ -0,0 +1,69 @@
+.import-preview__card {
+	max-width: 360px;
+	padding: rem( 16px );
```
Comment (jonathansadowski) [inline client/signup/steps/import-preview/style.scss]: I believe most of the steps have a `border-radius` of `3px`. Can we add that here for consistency?

Comment (andfinally) [reply client/signup/steps/import-preview/style.scss]: Thanks Jon, now added.

File: client/signup/steps/import-preview/index.jsx [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+			translate,
+		} = this.props;
+
+		let cardTitle = siteTitle;
+		let trimmedSiteUrl = importSiteUrl.replace( /^https?:\/\//i, '' );
+
+		if ( isEmpty( importSiteUrl ) ) {
+			// The user has an import file and told us the source
+			const engine = importSiteEngine === 'wordpress' ? 'WordPress' : startCase( importSiteEngine );
+			cardTitle = 'Importing from ' + engine;
```
Comment (jonathansadowski) [inline client/signup/steps/import-preview/index.jsx]: Can you wrap this in a translate? And can we use the config to get the engine title instead of calculating it from engine?

Comment (andfinally) [reply client/signup/steps/import-preview/index.jsx]: Thanks, that's better!

File: client/signup/steps/import-preview/index.jsx [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+
+		let cardTitle = siteTitle;
+		let trimmedSiteUrl = '';
+
+		if ( isEmpty( importSiteUrl ) ) {
+			// The user has an import file and told us the source
+			const importerData = importerConfig()[ importSiteEngine ];
+			cardTitle = translate( 'Importing from ' ) + importerData.title;
+		} else {
+			trimmedSiteUrl = importSiteUrl.replace( /^https?:\/\//i, '' ).replace( /\/$/, '' );
```
Comment (andfinally) [inline client/signup/steps/import-preview/index.jsx]: Too tired to make a combined regex pattern work right now, so I've done a double `replace`.

File: client/signup/steps/import-preview/index.jsx [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+			translate,
+		} = this.props;
+
+		let cardTitle = siteTitle;
+		let trimmedSiteUrl = '';
+
+		if ( isEmpty( importSiteUrl ) ) {
+			// The user has an import file and told us the source
+			const importerData = importerConfig()[ importSiteEngine ];
+			cardTitle = translate( 'Importing from ' ) + importerData.title;
```
Comment (jonathansadowski) [inline client/signup/steps/import-preview/index.jsx]: How about something like this?

```suggestion
			cardTitle = translate( 'Importing from %(importerName)s', {
				args: { importerName: importerData.title },
			} );
```

This makes the string more translatable in languages where the translation would need to put the engine name first.

File: client/signup/steps/import-preview/index.jsx (hunk 11 above)
Comment (jonathansadowski) [reply client/signup/steps/import-preview/index.jsx]: If you want to switch it, this should work, but I think it's also fine as two replaces:
```suggestion
			trimmedSiteUrl = importSiteUrl.replace( /^https?:\/\/(.*?)\/$/, '$1' );
```

File: client/signup/steps/import-preview/index.jsx [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+					{ translate( 'Continue' ) }
+				</Button>
+			</Card>
+		);
+	}
+
+	render() {
+		const { flowName, positionInFlow, stepName, translate } = this.props;
+
+		const headerText = translate( "Here's what we'll import to your new" + ' site.' );
```
Comment (p-jackson) [inline client/signup/steps/import-preview/index.jsx]: Maybe tidy up this string concat? My guess is it was just prettier going haywire.

File: client/signup/steps/import-preview/index.jsx (hunk 12 above)
Comment (andfinally) [reply client/signup/steps/import-preview/index.jsx]: Ahh much better, thanks!

File: client/signup/steps/import-preview/index.jsx (hunk 11 above)
Comment (andfinally) [reply client/signup/steps/import-preview/index.jsx]: I found I had to make the outer subpatterns optional, but now it works, thanks Jon.

File: client/signup/steps/import-preview/index.jsx (hunk 13 above)
Comment (andfinally) [reply client/signup/steps/import-preview/index.jsx]: Thanks Philip, I think it was - now tidied.

**Comment by andfinally** [comment]
Thanks for having a look @p-jackson! 😄 The behaviour on refresh is intentional – some of the flows, including this one, have the `disallowResume` parameter set, which returns you to the first step on refresh (the site type page in your case, since you're probably logged in). I'm not sure why we're doing this on some flows, but I suspect it's to ensure they hang on to all the data they need.

Your point about the flow from the main options on the site type list has saved us from introducing a regression, thank you! 🙏 I've added a condition in the `site-type` component that switches the flow back to onboarding when you select one of the main site types. It does feel a bit brittle checking for a particular flow in this condition, but I'm not sure there's a better way at this stage.
